### ISSUE TITLE
[Bug] Crashes when running `./craft migrate/all`

### ISSUE THREAD

**Issue by lindseydiloreto** [verbb/knock-knock#2]
It looks like Knock Knock needs to check if the request is a _console_ request before working its magic.

When I run this...
```
./craft migrate/all
```

It crashes...
```
Exception 'yii\base\UnknownMethodException' with message 'Calling unknown method: craft\console\Request::getUrl()'

in /.../vendor/yiisoft/yii2/base/Component.php:300

Stack trace:
#0 /.../vendor/verbb/knock-knock/src/KnockKnock.php(47): yii\base\Component->__call('getUrl', Array)
#1 /.../vendor/yiisoft/yii2/base/BaseObject.php(109): verbb\knockknock\KnockKnock->init()
#2 /.../vendor/yiisoft/yii2/base/Module.php(158): yii\base\BaseObject->__construct(Array)
#3 /.../vendor/craftcms/cms/src/base/Plugin.php(113): yii\base\Module->__construct('knock-knock', Object(craft\console\Application), Array)
#4 [internal function]: craft\base\Plugin->__construct('knock-knock', Object(craft\console\Application), Array)
#5 /.../vendor/yiisoft/yii2/di/Container.php(383): ReflectionClass->newInstanceArgs(Array)
#6 /.../vendor/yiisoft/yii2/di/Container.php(156): yii\di\Container->build('verbb\\knockknoc...', Array, Array)
#7 /.../vendor/yiisoft/yii2/BaseYii.php(349): yii\di\Container->get('verbb\\knockknoc...', Array, Array)
#8 /.../vendor/craftcms/cms/src/services/Plugins.php(789): yii\BaseYii::createObject(Array, Array)
#9 /.../vendor/craftcms/cms/src/services/Plugins.php(200): craft\services\Plugins->createPlugin('knock-knock', Array)
#10 /.../vendor/craftcms/cms/src/base/ApplicationTrait.php(1148): craft\services\Plugins->loadPlugins()
#11 /.../vendor/craftcms/cms/src/console/Application.php(64): craft\console\Application->_postInit()
#12 /.../vendor/yiisoft/yii2/base/BaseObject.php(109): craft\console\Application->init()
#13 /.../vendor/yiisoft/yii2/base/Application.php(206): yii\base\BaseObject->__construct(Array)
#14 /.../vendor/yiisoft/yii2/console/Application.php(89): yii\base\Application->__construct(Array)
#15 /.../vendor/craftcms/cms/src/console/Application.php(53): yii\console\Application->__construct(Array)
#16 [internal function]: craft\console\Application->__construct(Array)
#17 /.../vendor/yiisoft/yii2/di/Container.php(383): ReflectionClass->newInstanceArgs(Array)
#18 /.../vendor/yiisoft/yii2/di/Container.php(156): yii\di\Container->build('craft\\console\\A...', Array, Array)
#19 /.../vendor/yiisoft/yii2/BaseYii.php(349): yii\di\Container->get('craft\\console\\A...', Array, Array)
#20 /.../vendor/craftcms/cms/bootstrap/bootstrap.php(252): yii\BaseYii::createObject(Array)
#21 /.../vendor/craftcms/cms/bootstrap/console.php(31): require('/Users/lindseyd...')
#22 /.../craft(21): require('/Users/lindseyd...')
#23 {main}
```

Which results in this...
```
Script ./craft migrate/all handling the post-install-cmd event returned with error code 1
```

Disabling the plugin allows me to run `./craft migrate/all` with no problems.

**Comment by engram-design** [comment]
Fixed in [1.0.3](https://github.com/verbb/knock-knock/releases/tag/1.0.3)

**Comment by lindseydiloreto** [comment]
Brilliant, thanks man! Just updated my site, everything is working flawlessly. 🍺
